README.md

```diff
@@ -31,5 +40,30 @@ too; replace `-full` with `-oss` in any of the above commands to install the
 OSS distribution. Note that the default distribution and OSS distribution of
 a product can not be installed at the same time.
 
+## How do I ensure my configuration is up to date?
+
+Run the following command to update your configuration:
+
+    brew tap --repair elastic/tap
+    brew update -v
+
+Verify your configuration is based on the `main` branch with:
+
+    git -C /opt/homebrew/Library/Taps/elastic/homebrew-tap status
```

Comment:
The path seems to be `/usr/local/Homebrew/Library/Taps/elastic/homebrew-tap` on my macOS, however when running `brew tap --repair elastic/tap && brew update -v && git -C /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap status`, I still have master branch.

Reply:
I've done the branch switch manually using the following commands:
```
cd /usr/local/homebrew/Library/Taps/elastic/homebrew-tap
git fetch
git checkout main
```

After that, `brew update` doesn't complain anything

Reply:
In case this helps anyone else struggling or if we should consider refining the directions further for addressing this issue.

### Short (Solution)

If you have a single branch clone (aka shallow copy) of the `elastic-tap` repository, the best way to fix it is to [force an untap](https://docs.brew.sh/Manpage#untap---force-tap-) (so your packages stay installed) then re-tap it.

```zsh
brew untap elastic/tap --force
brew tap elastic/tap
```

### Long (Finding solution)

Alright this is what I have on my laptop
```zsh
% brew --version
Homebrew 3.3.9
Homebrew/homebrew-core (git revision fd70c7e700c; last commit 2022-01-07)
Homebrew/homebrew-cask (git revision aec4cc3d27; last commit 2022-01-08)
% git --version
git version 2.34.1
```

❌ [False fix 1st]  When I run the commands from the README.md update, it didn't fix the problem
```zsh
% brew tap --repair elastic/tap
% brew update
Updated 2 taps (homebrew/cask and caskroom/cask).
==> Updated Casks
powerphotos                trilium-notes              trilium-notes
Warning: Calling bottle :unneeded is deprecated! There is no replacement.
...
```

❌ [False fix 2nd] When I ran the alternate command from https://github.com/elastic/homebrew-tap/issues/110#issuecomment-984091527
```zsh
% brew update-reset $(brew --repo elastic/tap)
==> Fetching /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap...
error: Not a valid ref: refs/remotes/origin/main

==> Resetting /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.
```

Which leaves me at this
```zsh
% git -C /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

❌ [False fix 3rd] The alternate change of going into the repo did not work for me either,
```zsh
% cd /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap
% git fetch
% git checkout main
error: pathspec 'main' did not match any file(s) known to git
```

What is perplexing is the `HEAD` is `main` but it is not listed.
```zsh
% git remote show origin
* remote origin
  Fetch URL: https://github.com/elastic/homebrew-tap
  Push  URL: https://github.com/elastic/homebrew-tap
  HEAD branch: main
  Remote branch:
    master tracked
  Local branch configured for 'git pull':
    master merges with remote master
  Local ref configured for 'git push':
    master pushes to master (up to date)
```

😐 [Troubleshooting 1] However, a git clone has the main branch elsewhere works just fine
```zsh
% git clone https://github.com/elastic/homebrew-tap.git
Cloning into 'homebrew-tap'...
remote: Enumerating objects: 1258, done.
remote: Counting objects: 100% (499/499), done.
remote: Compressing objects: 100% (283/283), done.
remote: Total 1258 (delta 386), reused 279 (delta 215), pack-reused 759
Receiving objects: 100% (1258/1258), 304.36 KiB | 3.67 MiB/s, done.
Resolving deltas: 100% (972/972), done.
% git branch
* main
% git fetch --verbose
POST git-upload-pack (155 bytes)
From https://github.com/elastic/homebrew-tap
 = [up to date]      main            -> origin/main
 = [up to date]      main-as-default -> origin/main-as-default
 = [up to date]      marclop-v-prefix-removal -> origin/marclop-v-prefix-removal
 = [up to date]      master          -> origin/master
% git remote --verbose
origin	https://github.com/elastic/homebrew-tap.git (fetch)
origin	https://github.com/elastic/homebrew-tap.git (push)
```

❌ [False fix 4th] And resetting the URL doesn't fix the problem either.  I got the idea b/c git clone URL from github.com code was different.
```zsh
% git remote set-url origin https://github.com/elastic/homebrew-tap.git
% git remote show origin
* remote origin
  Fetch URL: https://github.com/elastic/homebrew-tap.git
  Push  URL: https://github.com/elastic/homebrew-tap.git
  HEAD branch: main
  Remote branch:
    master tracked
  Local branch configured for 'git pull':
    master merges with remote master
  Local ref configured for 'git push':
    master pushes to master (up to date)
% git remote --verbose
origin	https://github.com/elastic/homebrew-tap.git (fetch)
origin	https://github.com/elastic/homebrew-tap.git (push)
% git fetch origin --verbose
POST git-upload-pack (383 bytes)
From https://github.com/elastic/homebrew-tap
 = [up to date]      master     -> origin/master
 
 # Revert the change
 % git remote set-url origin https://github.com/elastic/homebrew-tap
```

😐 [Troubleshooting 2] I think `brew tap` when I set this up a while, cloned the repository as a [single branch](https://www.freecodecamp.org/news/git-clone-branch-how-to-clone-a-specific-branch/) (`git clone --branch <branchname> --single-branch <remote-repo-url>`) instead of all so the manual fix did not work for me. 💡 

```zsh
# Working, Main available
% git config --list --local
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
core.ignorecase=true
core.precomposeunicode=true
remote.origin.url=https://github.com/elastic/homebrew-tap.git
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
branch.main.remote=origin
branch.main.merge=refs/heads/main

# Broken, Main not available
% git config --list --local
core.repositoryformatversion=0
core.filemode=true
core.bare=false
core.logallrefupdates=true
core.ignorecase=true
core.precomposeunicode=true
core.autocrlf=false
remote.origin.url=https://github.com/elastic/homebrew-tap
remote.origin.fetch=+refs/heads/master:refs/remotes/origin/master
branch.master.remote=origin
branch.master.merge=refs/heads/master
homebrew.private=false
```

Found the related tickets and it is still the case that `brew update-reset` does not work on shallow clones
- https://github.com/Homebrew/brew/issues/7746
- https://github.com/Homebrew/brew/issues/7795

❌ [Failed fix 5th] Overriding the fetch branch
I just still ended up back on the master branch by the `brew update` command for some reason.

```zsh
% git config remote.origin.fetch
+refs/heads/master:refs/remotes/origin/master
% git remote set-branches origin main
% git config remote.origin.fetch
+refs/heads/main:refs/remotes/origin/main
% git fetch
From https://github.com/elastic/homebrew-tap
 * [new branch]      main       -> origin/main
 % git checkout main
Branch 'main' set up to track remote branch 'main' from 'origin'.
Switched to a new branch 'main'
% git branch -d master
error: The branch 'master' is not fully merged.
 % git branch -D master
Deleted branch master (was ab7b64e).
% git branch -rd origin/master
Deleted remote-tracking branch origin/master (was ab7b64e).
% brew upgrade
% git -C /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap status
On branch master
nothing to commit, working tree clean
```

💚 Solution: Force the untap so the existing packages stay installed, then re-tap it.
```zsh
% brew untap elastic/tap --force
Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/kibana-full.rb:9

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/filebeat-full.rb:10

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/logstash-full.rb:10

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/elasticsearch-full.rb:9

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/heartbeat-full.rb:10

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/auditbeat-full.rb:10

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/metricbeat-full.rb:10

Warning: Calling bottle :unneeded is deprecated! There is no replacement.
Please report this issue to the elastic/tap tap (not Homebrew/brew or Homebrew/core):
  /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap/Formula/packetbeat-full.rb:10

Warning: Untapping elastic/tap even though it contains the following installed formulae or casks:
kibana-full
filebeat-full
logstash-full
elasticsearch-full
heartbeat-full
auditbeat-full
metricbeat-full
packetbeat-full
Untapping elastic/tap...
Untapped 17 formulae (1,241 files, 1.3MB).

% brew tap elastic/tap
==> Tapping elastic/tap
Cloning into '/usr/local/Homebrew/Library/Taps/elastic/homebrew-tap'...
remote: Enumerating objects: 1258, done.
remote: Counting objects: 100% (499/499), done.
remote: Compressing objects: 100% (283/283), done.
remote: Total 1258 (delta 386), reused 279 (delta 215), pack-reused 759
Receiving objects: 100% (1258/1258), 304.36 KiB | 2.98 MiB/s, done.
Resolving deltas: 100% (972/972), done.
Tapped 17 formulae (37 files, 413.9KB).

% brew outdated
elastic/tap/auditbeat-full (7.15.2) < 7.16.2
elastic/tap/elasticsearch-full (7.15.2) < 7.16.2
elastic/tap/filebeat-full (7.15.2) < 7.16.2
elastic/tap/heartbeat-full (7.15.2) < 7.16.2
elastic/tap/kibana-full (7.15.2) < 7.16.2
elastic/tap/logstash-full (7.15.2) < 7.16.2
elastic/tap/metricbeat-full (7.15.2) < 7.16.2
elastic/tap/packetbeat-full (7.15.2) < 7.16.2
% brew update
Already up-to-date.
% brew upgrade
==> Upgrading 8 outdated packages:
elastic/tap/kibana-full 7.15.2 -> 7.16.2
elastic/tap/filebeat-full 7.15.2 -> 7.16.2
elastic/tap/logstash-full 7.15.2 -> 7.16.2
elastic/tap/elasticsearch-full 7.15.2 -> 7.16.2
elastic/tap/heartbeat-full 7.15.2 -> 7.16.2
elastic/tap/auditbeat-full 7.15.2 -> 7.16.2
elastic/tap/metricbeat-full 7.15.2 -> 7.16.2
elastic/tap/packetbeat-full 7.15.2 -> 7.16.2
==> Downloading https://artifacts.elastic.co/downloads/beats/packetbeat/packetbe
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/beats/filebeat/filebeat-7
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/logstash/logstash-7.16.2-
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/elasticsearch/elasticsear
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/beats/heartbeat/heartbeat
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/beats/auditbeat/auditbeat
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/beats/metricbeat/metricbe
######################################################################## 100.0%
==> Downloading https://artifacts.elastic.co/downloads/kibana/kibana-7.16.2-darw
######################################################################## 100.0%
==> Upgrading elastic/tap/packetbeat-full
  7.15.2 -> 7.16.2

==> Caveats
To start elastic/tap/packetbeat-full now and restart at login:
  brew services start elastic/tap/packetbeat-full
Or, if you don't want/need a background service you can just run:
  packetbeat
==> Summary
🍺  /usr/local/Cellar/packetbeat-full/7.16.2: 139 files, 88.0MB, built in 5 seconds
==> Running `brew cleanup packetbeat-full`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /usr/local/Cellar/packetbeat-full/7.15.2... (139 files, 89MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/packetbeat-full--7.15.2.tar.gz... (22.9MB)
==> Upgrading elastic/tap/filebeat-full
  7.15.2 -> 7.16.2

==> Caveats
To start elastic/tap/filebeat-full now and restart at login:
  brew services start elastic/tap/filebeat-full
Or, if you don't want/need a background service you can just run:
  filebeat
==> Summary
🍺  /usr/local/Cellar/filebeat-full/7.16.2: 1,412 files, 115.3MB, built in 7 seconds
==> Running `brew cleanup filebeat-full`...
Removing: /usr/local/Cellar/filebeat-full/7.15.2... (1,413 files, 116.8MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/filebeat-full--7.15.2.tar.gz... (28.8MB)
==> Upgrading elastic/tap/logstash-full
  7.15.2 -> 7.16.2

==> find /usr/local/Cellar/logstash-full/7.16.2/libexec/jdk.app/Contents/Home/bi
==> Caveats
Please read the getting started guide located at:
  https://www.elastic.co/guide/en/logstash/current/getting-started-with-logstash.html

To start elastic/tap/logstash-full now and restart at login:
  brew services start elastic/tap/logstash-full
Or, if you don't want/need a background service you can just run:
  logstash
==> Summary
🍺  /usr/local/Cellar/logstash-full/7.16.2: 13,870 files, 605.2MB, built in 32 seconds
==> Running `brew cleanup logstash-full`...
Removing: /usr/local/Cellar/logstash-full/7.15.2... (14,135 files, 613.9MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/logstash-full--7.15.2.tar.gz... (354.9MB)
==> Upgrading elastic/tap/elasticsearch-full
  7.15.2 -> 7.16.2

Warning: Tried to install empty array to /usr/local/etc/elasticsearch/jvm.options.d
==> codesign -f -s - /usr/local/Cellar/elasticsearch-full/7.16.2/libexec/modules
==> find /usr/local/Cellar/elasticsearch-full/7.16.2/libexec/jdk.app/Contents/Ho
==> Caveats
Data:    /usr/local/var/lib/elasticsearch/elasticsearch_nalbee/
Logs:    /usr/local/var/log/elasticsearch/elasticsearch_nalbee.log
Plugins: /usr/local/var/elasticsearch/plugins/
Config:  /usr/local/etc/elasticsearch/

To start elastic/tap/elasticsearch-full now and restart at login:
  brew services start elastic/tap/elasticsearch-full
Or, if you don't want/need a background service you can just run:
  elasticsearch
==> Summary
🍺  /usr/local/Cellar/elasticsearch-full/7.16.2: 974 files, 530.6MB, built in 10 seconds
==> Running `brew cleanup elasticsearch-full`...
Removing: /usr/local/Cellar/elasticsearch-full/7.15.2... (974 files, 527.1MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/elasticsearch-full--7.15.2.tar.gz... (323.0MB)
==> Upgrading elastic/tap/heartbeat-full
  7.15.2 -> 7.16.2

==> Caveats
To start elastic/tap/heartbeat-full now and restart at login:
  brew services start elastic/tap/heartbeat-full
Or, if you don't want/need a background service you can just run:
  heartbeat
==> Summary
🍺  /usr/local/Cellar/heartbeat-full/7.16.2: 9 files, 85.5MB, built in 4 seconds
==> Running `brew cleanup heartbeat-full`...
Removing: /usr/local/Cellar/heartbeat-full/7.15.2... (9 files, 86.2MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/heartbeat-full--7.15.2.tar.gz... (22MB)
==> Upgrading elastic/tap/auditbeat-full
  7.15.2 -> 7.16.2

==> Caveats
To start elastic/tap/auditbeat-full now and restart at login:
  brew services start elastic/tap/auditbeat-full
Or, if you don't want/need a background service you can just run:
  auditbeat
==> Summary
🍺  /usr/local/Cellar/auditbeat-full/7.16.2: 95 files, 85.8MB, built in 5 seconds
==> Running `brew cleanup auditbeat-full`...
Removing: /usr/local/Cellar/auditbeat-full/7.15.2... (95 files, 86.8MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/auditbeat-full--7.15.2.tar.gz... (22.1MB)
==> Upgrading elastic/tap/metricbeat-full
  7.15.2 -> 7.16.2

==> Caveats
To start elastic/tap/metricbeat-full now and restart at login:
  brew services start elastic/tap/metricbeat-full
Or, if you don't want/need a background service you can just run:
  metricbeat
==> Summary
🍺  /usr/local/Cellar/metricbeat-full/7.16.2: 1,007 files, 141.1MB, built in 7 seconds
==> Running `brew cleanup metricbeat-full`...
Removing: /usr/local/Cellar/metricbeat-full/7.15.2... (1,004 files, 139.3MB)
Removing: /Users/nalbee/Library/Caches/Homebrew/metricbeat-full--7.15.2.tar.gz... (33.7MB)
==> Upgrading elastic/tap/kibana-full
  7.15.2 -> 7.16.2

==> Caveats
Config: /usr/local/etc/kibana/
If you wish to preserve your plugins upon upgrade, make a copy of
/usr/local/opt/kibana-full/plugins before upgrading, and copy it into the
new keg location after upgrading.

To start elastic/tap/kibana-full now and restart at login:
  brew services start elastic/tap/kibana-full
Or, if you don't want/need a background service you can just run:
  kibana
==> Summary
🍺  /usr/local/Cellar/kibana-full/7.16.2: 51,328 files, 716.9MB, built in 1 minute 25 seconds
==> Running `brew cleanup kibana-full`...
Removing: /usr/local/Cellar/kibana-full/7.15.2... (49,345 files, 1GB)
Removing: /Users/nalbee/Library/Caches/Homebrew/kibana-full--7.15.2.tar.gz... (269.5MB)
==> Caveats
==> packetbeat-full
To start elastic/tap/packetbeat-full now and restart at login:
  brew services start elastic/tap/packetbeat-full
Or, if you don't want/need a background service you can just run:
  packetbeat
==> filebeat-full
To start elastic/tap/filebeat-full now and restart at login:
  brew services start elastic/tap/filebeat-full
Or, if you don't want/need a background service you can just run:
  filebeat
==> logstash-full
Please read the getting started guide located at:
  https://www.elastic.co/guide/en/logstash/current/getting-started-with-logstash.html

To start elastic/tap/logstash-full now and restart at login:
  brew services start elastic/tap/logstash-full
Or, if you don't want/need a background service you can just run:
  logstash
==> elasticsearch-full
Data:    /usr/local/var/lib/elasticsearch/elasticsearch_nalbee/
Logs:    /usr/local/var/log/elasticsearch/elasticsearch_nalbee.log
Plugins: /usr/local/var/elasticsearch/plugins/
Config:  /usr/local/etc/elasticsearch/

To start elastic/tap/elasticsearch-full now and restart at login:
  brew services start elastic/tap/elasticsearch-full
Or, if you don't want/need a background service you can just run:
  elasticsearch
==> heartbeat-full
To start elastic/tap/heartbeat-full now and restart at login:
  brew services start elastic/tap/heartbeat-full
Or, if you don't want/need a background service you can just run:
  heartbeat
==> auditbeat-full
To start elastic/tap/auditbeat-full now and restart at login:
  brew services start elastic/tap/auditbeat-full
Or, if you don't want/need a background service you can just run:
  auditbeat
==> metricbeat-full
To start elastic/tap/metricbeat-full now and restart at login:
  brew services start elastic/tap/metricbeat-full
Or, if you don't want/need a background service you can just run:
  metricbeat
==> kibana-full
Config: /usr/local/etc/kibana/
If you wish to preserve your plugins upon upgrade, make a copy of
/usr/local/opt/kibana-full/plugins before upgrading, and copy it into the
new keg location after upgrading.

To start elastic/tap/kibana-full now and restart at login:
  brew services start elastic/tap/kibana-full
Or, if you don't want/need a background service you can just run:
  kibana
```

Success at last. 😄

Reply:
Thanks @a03nikki @jmlrt I pushed e5bde99f43e122848300f8c607e4ae46979e6d69 
please take another look.